pkg/controllers/nodeclass/status/readiness.go

```diff
@@ -41,5 +43,11 @@ func (n Readiness) Reconcile(ctx context.Context, nodeClass *v1.EC2NodeClass) (r
 			return reconcile.Result{}, fmt.Errorf("failed to detect the cluster CIDR, %w", err)
 		}
 	}
+	if _, found := lo.FindKeyBy(nodeClass.Spec.Tags, func(key string, _ string) bool {
+		return strings.HasPrefix(key, "kubernetes.io/cluster/")
+	}); found {
+		nodeClass.StatusConditions().SetFalse(status.ConditionReady, "NodeClassNotReady", "validation bypassed")
```

Comment:
We generally don't set the Readiness condition directly since it doesn't have direct ownership and can be updated by the readiness of other conditions -- I think we may need another condition

Reply:
We have a validation condition on NodePool: https://github.com/kubernetes-sigs/karpenter/blob/main/pkg/apis/v1/nodepool_status.go#L27